tests/tests_pytorch/utilities/migration/test_migration.py

```diff
@@ -0,0 +1,111 @@
+from pytorch_lightning.utilities.migration import migrate_checkpoint
+from pytorch_lightning.utilities.migration.utils import _get_version, _set_legacy_version, _set_version
+
+
+@pytest.mark.parametrize(
+    "old_checkpoint, new_checkpoint",
+    [
+        (
+            {"epoch": 1, "global_step": 23, "checkpoint_callback_best": 0.34},
+            {"epoch": 1, "global_step": 23, "callbacks": {ModelCheckpoint: {"best_model_score": 0.34}}, "loops": ANY},
+        ),
+        (
+            {"epoch": 1, "global_step": 23, "checkpoint_callback_best_model_score": 0.99},
+            {"epoch": 1, "global_step": 23, "callbacks": {ModelCheckpoint: {"best_model_score": 0.99}}, "loops": ANY},
+        ),
+        (
+            {"epoch": 1, "global_step": 23, "checkpoint_callback_best_model_path": "path"},
+            {"epoch": 1, "global_step": 23, "callbacks": {ModelCheckpoint: {"best_model_path": "path"}}, "loops": ANY},
+        ),
+        (
+            {"epoch": 1, "global_step": 23, "early_stop_callback_wait": 2, "early_stop_callback_patience": 4},
+            {
+                "epoch": 1,
+                "global_step": 23,
+                "callbacks": {EarlyStopping: {"wait_count": 2, "patience": 4}},
+                "loops": ANY,
+            },
+        ),
+    ],
+)
+def test_migrate_model_checkpoint_early_stopping(tmpdir, old_checkpoint, new_checkpoint):
+    _set_version(old_checkpoint, "0.9.0")
+    _set_legacy_version(new_checkpoint, "0.9.0")
+    _set_version(new_checkpoint, pl.__version__)
+    updated_checkpoint, _ = migrate_checkpoint(old_checkpoint)
+    assert updated_checkpoint == old_checkpoint == new_checkpoint
+    assert _get_version(updated_checkpoint) == pl.__version__
+
+
+def test_migrate_loop_global_step_to_progress_tracking():
+    old_checkpoint = {"global_step": 15, "epoch": 2}
+    _set_version(old_checkpoint, "1.5.9")  # pretend a checkpoint prior to 1.6.0
+    updated_checkpoint, _ = migrate_checkpoint(old_checkpoint)
+    # automatic optimization
+    assert (
+        updated_checkpoint["loops"]["fit_loop"]["epoch_loop.batch_loop.optimizer_loop.optim_progress"]["optimizer"][
+            "step"
+        ]["total"]["completed"]
+        == 15
+    )
+    # for manual optimization
+    assert (
+        updated_checkpoint["loops"]["fit_loop"]["epoch_loop.batch_loop.manual_loop.optim_step_progress"]["total"][
+            "completed"
+        ]
+        == 15
+    )
+
+
+def test_migrate_loop_current_epoch_to_progress_tracking():
+    old_checkpoint = {"global_step": 15, "epoch": 2}
+    _set_version(old_checkpoint, "1.5.9")  # pretend a checkpoint prior to 1.6.0
+    updated_checkpoint, _ = migrate_checkpoint(old_checkpoint)
+    assert updated_checkpoint["loops"]["fit_loop"]["epoch_progress"]["current"]["completed"] == 2
+
+
+@pytest.mark.parametrize("model_class", [BoringModel, ManualOptimBoringModel])
+def test_migrate_loop_batches_that_stepped(tmpdir, model_class):
```

Comment:
only change to this test is the name and inserted `_set_version(ckpt, "1.6.4")` plus adjusted comment